python/ray/tune/hyperband.py

```diff
@@ -30,33 +30,53 @@ class HyperBandScheduler(FIFOScheduler):
 
     Trials added will be inserted into the most recent bracket
     and band and will spill over to new brackets/bands accordingly.
+
+    See https://people.eecs.berkeley.edu/~kjamieson/hyperband.html
+
+    Args:
+        time_attr (str): The TrainingResult attr to use for comparing time.
+            Note that you can pass in something non-temporal such as
+            `training_iteration` as a measure of progress, the only requirement
+            is that the attribute should increase monotonically.
+        reward_attr (str): The TrainingResult objective value attribute. As
+            with `time_attr`, this may refer to any objective value. Stopping
+            procedures will use this attribute.
+        max_t_attr (int): maximum iterations per configuration. Altering
+            this with `eta=None` will not change the bracket count
+            nor size. If eta is changed to an int, then all will change.
+        eta (int): Default is None, This maintains the bracket size
```

Comment:
This is kind of confusing, how about we just set the default to 3 (configurable), and always enforce num brackets to 5. There is no legit reason to have hundreds of brackets or anything like that.

Reply:
Sounds good

Reply:
Although if you said that eta is 4, I don't know how many trials to put in each of the 5 brackets. Number of brackets is determined by eta.

Reply:
I see, so if eta is 4 or if max_t is smaller you would have less than 5 brackets? Maybe that's ok. We could change num_brackets to a max_brackets param to make it clear. So the options would be
- max_t (user must define)
- eta (defaults to 3)
- max_brackets (defaults to 5)

Reply:
Not necessarily. 

I think the big question here is how close do you want to be to the original HyperBand formulation, and if we even want to expose it?  

Right now, if eta and max_t is passed in, then we use the HyperBand formulation. 
If `max_brackets` is a parameter, it's a complete departure from any of the theory of setting brackets, number of trials per bracket, etc etc. In which case, the title "HyperBand" is misleading and we should just call it "SuccessiveHalving".

Reply:
As I understand it, the original formulation assumes your units are reasonable (i.e. 1 is a reasonable minimum resources allocation). However in our case our units have arbitrary scales, so we need to squash them down. We do this by setting max_brackets, which should be similar to rescaling the units until the number of brackets is few enough.

For distribution -- can't you just allocate 1 trial to the largest bracket, and proceed from there? Excess trials spill over to new rounds.